internal/garbagecollector/garbageCollector.go

```diff
@@ -0,0 +1,91 @@
+package garbagecollector
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/jinzhu/gorm"
+	clusterPkg "github.com/openshift/assisted-service/internal/cluster"
+	"github.com/openshift/assisted-service/internal/host"
+	"github.com/openshift/assisted-service/pkg/leader"
+	"github.com/openshift/assisted-service/pkg/s3wrapper"
+	"github.com/openshift/assisted-service/restapi/operations/installer"
+	"github.com/sirupsen/logrus"
+)
+
+type Config struct {
+	DeletedUnregisteredAfter time.Duration `envconfig:"DELETED_UNREGISTERED_AFTER" default:"168h"` // 7d
+	DeregisterInactiveAfter  time.Duration `envconfig:"DELETED_INACTIVE_AFTER" default:"720h"`     // 20d
+}
+
+type GarbageCollectors interface {
+	PermanentClustersDeletion(ctx context.Context, olderThan strfmt.DateTime, objectHandler s3wrapper.API) error
+	DeregisterClusterInternal(ctx context.Context, params installer.DeregisterClusterParams) error
+}
+
+func NewGarbageCollectors(
```

Comment:
You don't set the config on creation so  `DeletedUnregisteredAfter` and `DeregisterInactiveAfter` is 0